test/integration/targets/ansible-galaxy-collection/tasks/install_offline.yml

```
@@ -40,6 +39,93 @@
     - name: install the tarfile with the installed dependency
       command: ansible-galaxy collection install {{ build_dir }}/ns-coll1-1.0.0.tar.gz -p {{ install_dir }} -s offline
 
+    - name: empty the installed collections directory
+      file:
+        path: "{{ install_dir }}"
+        state: "{{ item }}"
+      loop:
+        - absent
+        - directory
+
+    - name: edit skeleton with new versions to test upgrading
+      lineinfile:
+        path: "{{ galaxy_dir }}/offline/setup/ns/{{ item }}/galaxy.yml"
+        regexp: "^version:.*$"
+        line: "version: 2.0.0"
+      loop:
+        - coll1
+        - coll2
+
+    - name: build the tarfiles to test upgrading
+      command: ansible-galaxy collection build {{ init_dir }}/ns/{{ item }}
+      args:
+        chdir: "{{ build_dir }}"
+      loop:
+        - coll1
+        - coll2
+
+    - name: install the tarfile and its dep with an unreachable server (expected error)
+      command: ansible-galaxy collection install {{ build_dir }}/ns-coll1-1.0.0.tar.gz -p {{ install_dir }} -s offline
+      environment:
+        ANSIBLE_FORCE_COLOR: False
+        ANSIBLE_NOCOLOR: True
+      ignore_errors: yes
+      register: missing_dep
+
+    - name: install the tarfile with a missing dependency and --offline
+      command: ansible-galaxy collection install {{ build_dir }}/ns-coll1-1.0.0.tar.gz -p {{ install_dir }} -s offline --offline
```

That `offline --offline` got me puzzled for a minute. Let's try to avoid such brain tricks :)
```suggestion
      command: ansible-galaxy collection install --offline {{ build_dir }}/ns-coll1-1.0.0.tar.gz -p {{ install_dir }} -s offline
```